From 9e332e6cb2ab06aa9a6e8770a70bcd884f904b55 Mon Sep 17 00:00:00 2001
From: Alejandro Sierra <b18039@freescale.com>
Date: Tue, 19 Jan 2016 19:26:27 -0600
Subject: [PATCH 24/37] ARM: dts: add fix & interleave mode imx6dqscm

Support added for fix interleave mode on LPDDR2. IMX6DQSCM platforms:
    - imx6dqscm-1gb-evb
    - imx6dqscm-2gb-evb
    - imx6dqscm-1gb-qwks

Signed-off-by: Alejandro Sierra <b18039@freescale.com>
---
 arch/arm/boot/dts/Makefile                         |  42 +++--
 .../boot/dts/imx6dqscm-1gb-evb-btwifi-fix-ldo.dts  |  10 +
 .../imx6dqscm-1gb-evb-btwifi-interleave-ldo.dts    |  10 +
 arch/arm/boot/dts/imx6dqscm-1gb-evb-btwifi-ldo.dts |  10 -
 .../boot/dts/imx6dqscm-1gb-evb-enetirq-fix-ldo.dts |  18 ++
 .../imx6dqscm-1gb-evb-enetirq-interleave-ldo.dts   |  18 ++
 .../arm/boot/dts/imx6dqscm-1gb-evb-enetirq-ldo.dts |  18 --
 arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts    | 202 +++++++++++++++++++++
 .../boot/dts/imx6dqscm-1gb-evb-hdcp-fix-ldo.dts    |  23 +++
 .../dts/imx6dqscm-1gb-evb-hdcp-interleave-ldo.dts  |  23 +++
 arch/arm/boot/dts/imx6dqscm-1gb-evb-hdcp-ldo.dts   |  23 ---
 .../boot/dts/imx6dqscm-1gb-evb-interleave-ldo.dts  | 194 ++++++++++++++++++++
 arch/arm/boot/dts/imx6dqscm-1gb-evb-ldo.dts        | 194 --------------------
 arch/arm/boot/dts/imx6dqscm-1gb-fix.dtsi           |  18 ++
 arch/arm/boot/dts/imx6dqscm-1gb-interleave.dtsi    |  19 ++
 .../boot/dts/imx6dqscm-1gb-qwks-rev1-fix-ldo.dts   |  67 +++++++
 .../dts/imx6dqscm-1gb-qwks-rev1-hdcp-fix-ldo.dts   |  23 +++
 ...imx6dqscm-1gb-qwks-rev1-hdcp-interleave-ldo.dts |  23 +++
 .../boot/dts/imx6dqscm-1gb-qwks-rev1-hdcp-ldo.dts  |  23 ---
 .../dts/imx6dqscm-1gb-qwks-rev1-interleave-ldo.dts |  67 +++++++
 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-ldo.dts  |  67 -------
 .../dts/imx6dqscm-1gb-qwks-rev1-wifi-fix-ldo.dts   |  10 +
 ...imx6dqscm-1gb-qwks-rev1-wifi-interleave-ldo.dts |  10 +
 .../boot/dts/imx6dqscm-1gb-qwks-rev1-wifi-ldo.dts  |  10 -
 .../boot/dts/imx6dqscm-1gb-qwks-rev2-fix-ldo.dts   |  68 +++++++
 .../dts/imx6dqscm-1gb-qwks-rev2-hdcp-fix-ldo.dts   |  23 +++
 ...imx6dqscm-1gb-qwks-rev2-hdcp-interleave-ldo.dts |  23 +++
 .../boot/dts/imx6dqscm-1gb-qwks-rev2-hdcp-ldo.dts  |  23 ---
 .../dts/imx6dqscm-1gb-qwks-rev2-interleave-ldo.dts |  68 +++++++
 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-ldo.dts  |  67 -------
 .../dts/imx6dqscm-1gb-qwks-rev2-wifi-fix-ldo.dts   |  10 +
 ...imx6dqscm-1gb-qwks-rev2-wifi-interleave-ldo.dts |  10 +
 .../boot/dts/imx6dqscm-1gb-qwks-rev2-wifi-ldo.dts  |  10 -
 .../boot/dts/imx6dqscm-2gb-evb-btwifi-fix-ldo.dts  |  15 ++
 .../imx6dqscm-2gb-evb-btwifi-interleave-ldo.dts    |  14 ++
 arch/arm/boot/dts/imx6dqscm-2gb-evb-btwifi-ldo.dts |  21 ---
 .../boot/dts/imx6dqscm-2gb-evb-enetirq-fix-ldo.dts |  15 ++
 .../imx6dqscm-2gb-evb-enetirq-interleave-ldo.dts   |  14 ++
 .../arm/boot/dts/imx6dqscm-2gb-evb-enetirq-ldo.dts |  21 ---
 arch/arm/boot/dts/imx6dqscm-2gb-evb-fix-ldo.dts    |  15 ++
 .../boot/dts/imx6dqscm-2gb-evb-hdcp-fix-ldo.dts    |  15 ++
 .../dts/imx6dqscm-2gb-evb-hdcp-interleave-ldo.dts  |  15 ++
 arch/arm/boot/dts/imx6dqscm-2gb-evb-hdcp-ldo.dts   |  21 ---
 .../boot/dts/imx6dqscm-2gb-evb-interleave-ldo.dts  |  15 ++
 arch/arm/boot/dts/imx6dqscm-2gb-evb-ldo.dts        |  21 ---
 arch/arm/boot/dts/imx6dqscm-2gb-fix.dtsi           |  18 ++
 arch/arm/boot/dts/imx6dqscm-2gb-interleave.dtsi    |  19 ++
 47 files changed, 1120 insertions(+), 543 deletions(-)
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-evb-btwifi-fix-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-evb-btwifi-interleave-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-evb-btwifi-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-evb-enetirq-fix-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-evb-enetirq-interleave-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-evb-enetirq-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-evb-hdcp-fix-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-evb-hdcp-interleave-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-evb-hdcp-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-evb-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-fix.dtsi
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-interleave.dtsi
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-fix-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-hdcp-fix-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-hdcp-interleave-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-hdcp-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-interleave-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-wifi-fix-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-wifi-interleave-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-wifi-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-fix-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-hdcp-fix-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-hdcp-interleave-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-hdcp-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-interleave-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-wifi-fix-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-wifi-interleave-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-wifi-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-2gb-evb-btwifi-fix-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-2gb-evb-btwifi-interleave-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6dqscm-2gb-evb-btwifi-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-2gb-evb-enetirq-fix-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-2gb-evb-enetirq-interleave-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6dqscm-2gb-evb-enetirq-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-2gb-evb-fix-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-2gb-evb-hdcp-fix-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-2gb-evb-hdcp-interleave-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6dqscm-2gb-evb-hdcp-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-2gb-evb-interleave-ldo.dts
 delete mode 100644 arch/arm/boot/dts/imx6dqscm-2gb-evb-ldo.dts
 create mode 100644 arch/arm/boot/dts/imx6dqscm-2gb-fix.dtsi
 create mode 100644 arch/arm/boot/dts/imx6dqscm-2gb-interleave.dtsi

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index e411d87..cf95845 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -242,20 +242,34 @@ dtb-$(CONFIG_ARCH_MXC) += \
 	vf610-cosmic.dtb \
 	imx6sx-sdb-emmc.dtb \
 	imx6sx-sdb-m4.dtb \
-	imx6dqscm-1gb-evb-ldo.dtb \
-	imx6dqscm-2gb-evb-ldo.dtb \
-	imx6dqscm-1gb-evb-btwifi-ldo.dtb \
-	imx6dqscm-2gb-evb-btwifi-ldo.dtb \
-	imx6dqscm-1gb-evb-enetirq-ldo.dtb \
-	imx6dqscm-2gb-evb-enetirq-ldo.dtb \
-	imx6dqscm-1gb-evb-hdcp-ldo.dtb \
-	imx6dqscm-2gb-evb-hdcp-ldo.dtb \
-	imx6dqscm-1gb-qwks-rev1-ldo.dtb \
-	imx6dqscm-1gb-qwks-rev1-hdcp-ldo.dtb \
-	imx6dqscm-1gb-qwks-rev1-wifi-ldo.dtb \
-        imx6dqscm-1gb-qwks-rev2-ldo.dtb \
-        imx6dqscm-1gb-qwks-rev2-hdcp-ldo.dtb \
-        imx6dqscm-1gb-qwks-rev2-wifi-ldo.dtb \
+	imx6dqscm-1gb-evb-fix-ldo.dtb \
+	imx6dqscm-1gb-evb-interleave-ldo.dtb \
+	imx6dqscm-1gb-evb-btwifi-fix-ldo.dtb \
+	imx6dqscm-1gb-evb-btwifi-interleave-ldo.dtb \
+	imx6dqscm-1gb-evb-enetirq-fix-ldo.dtb \
+	imx6dqscm-1gb-evb-enetirq-interleave-ldo.dtb \
+	imx6dqscm-1gb-evb-hdcp-fix-ldo.dtb \
+	imx6dqscm-1gb-evb-hdcp-interleave-ldo.dtb \
+	imx6dqscm-2gb-evb-fix-ldo.dtb \
+        imx6dqscm-2gb-evb-interleave-ldo.dtb \
+        imx6dqscm-2gb-evb-btwifi-fix-ldo.dtb \
+        imx6dqscm-2gb-evb-btwifi-interleave-ldo.dtb \
+        imx6dqscm-2gb-evb-enetirq-fix-ldo.dtb \
+        imx6dqscm-2gb-evb-enetirq-interleave-ldo.dtb \
+        imx6dqscm-2gb-evb-hdcp-fix-ldo.dtb \
+        imx6dqscm-2gb-evb-hdcp-interleave-ldo.dtb \
+	imx6dqscm-1gb-qwks-rev1-fix-ldo.dtb \
+        imx6dqscm-1gb-qwks-rev1-interleave-ldo.dtb \
+        imx6dqscm-1gb-qwks-rev1-wifi-fix-ldo.dtb \
+        imx6dqscm-1gb-qwks-rev1-wifi-interleave-ldo.dtb \
+	imx6dqscm-1gb-qwks-rev1-hdcp-fix-ldo.dtb \
+        imx6dqscm-1gb-qwks-rev1-hdcp-interleave-ldo.dtb \
+	imx6dqscm-1gb-qwks-rev2-fix-ldo.dtb \
+        imx6dqscm-1gb-qwks-rev2-interleave-ldo.dtb \
+        imx6dqscm-1gb-qwks-rev2-wifi-fix-ldo.dtb \
+        imx6dqscm-1gb-qwks-rev2-wifi-interleave-ldo.dtb \
+        imx6dqscm-1gb-qwks-rev2-hdcp-fix-ldo.dtb \
+        imx6dqscm-1gb-qwks-rev2-hdcp-interleave-ldo.dtb \
 	vf610-twr.dtb \
 	imx7d-12x12-lpddr3-arm2.dtb \
 	imx7d-12x12-ddr3-arm2.dtb \
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-btwifi-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-btwifi-fix-ldo.dts
new file mode 100644
index 0000000..0880735
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-evb-btwifi-fix-ldo.dts
@@ -0,0 +1,10 @@
+/*
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6dqscm-1gb-evb-fix-ldo.dts"
+#include "imx6qdl-sabresd-btwifi.dtsi"
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-btwifi-interleave-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-btwifi-interleave-ldo.dts
new file mode 100644
index 0000000..7090308
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-evb-btwifi-interleave-ldo.dts
@@ -0,0 +1,10 @@
+/*
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6dqscm-1gb-evb-interleave-ldo.dts"
+#include "imx6qdl-sabresd-btwifi.dtsi"
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-btwifi-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-btwifi-ldo.dts
deleted file mode 100644
index 800ad85..0000000
--- a/arch/arm/boot/dts/imx6dqscm-1gb-evb-btwifi-ldo.dts
+++ /dev/null
@@ -1,10 +0,0 @@
-/*
- * Copyright (C) 2015 Freescale Semiconductor, Inc.
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- */
-
-#include "imx6dqscm-1gb-evb-ldo.dts"
-#include "imx6qdl-sabresd-btwifi.dtsi"
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-enetirq-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-enetirq-fix-ldo.dts
new file mode 100644
index 0000000..bd748f1
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-evb-enetirq-fix-ldo.dts
@@ -0,0 +1,18 @@
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6dqscm-1gb-evb-fix-ldo.dts"
+
+&fec {
+	pinctrl-0 = <&pinctrl_enet &pinctrl_enet_irq>;
+	interrupts-extended = <&gpio1 6 0x04>, <&intc 0 119 0x04>;
+};
+
+&i2c3 {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-enetirq-interleave-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-enetirq-interleave-ldo.dts
new file mode 100644
index 0000000..5bf21ec
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-evb-enetirq-interleave-ldo.dts
@@ -0,0 +1,18 @@
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6dqscm-1gb-evb-interleave-ldo.dts"
+
+&fec {
+	pinctrl-0 = <&pinctrl_enet &pinctrl_enet_irq>;
+	interrupts-extended = <&gpio1 6 0x04>, <&intc 0 119 0x04>;
+};
+
+&i2c3 {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-enetirq-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-enetirq-ldo.dts
deleted file mode 100644
index b11c94d..0000000
--- a/arch/arm/boot/dts/imx6dqscm-1gb-evb-enetirq-ldo.dts
+++ /dev/null
@@ -1,18 +0,0 @@
-/*
- * Copyright (C) 2014 Freescale Semiconductor, Inc.
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- */
-
-#include "imx6dqscm-1gb-evb-ldo.dts"
-
-&fec {
-	pinctrl-0 = <&pinctrl_enet &pinctrl_enet_irq>;
-	interrupts-extended = <&gpio1 6 0x04>, <&intc 0 119 0x04>;
-};
-
-&i2c3 {
-	status = "disabled";
-};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts
new file mode 100644
index 0000000..9ccf1ee
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6q-sabresd-ldo.dts"
+
+/ {
+	model = "Freescale i.MX6D SCM EVB";
+	compatible = "fsl,imx6q-sabresd", "fsl,imx6q";
+	memory: memory {
+               linux,usable-memory = <0x10000000 0x20000000 0x80000000 0x20000000>; 
+        };
+        soc {
+            busfreq {
+                status = "disabled";
+            };
+        };
+};
+
+&ecspi1 {
+	cs-gpios = <&gpio2 30 0>;
+	internal_scm_flash: m25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "micron,n25q128a13";
+		spi-max-frequency = <20000000>;
+		reg = <0>;
+	};
+};
+
+&i2c1 {
+	ov564x: ov564x@3c {
+		DOVDD-supply = <&sw4_reg>; /* 1.8v */
+	};
+};
+
+&clks {
+	fsl,ldb-di0-parent = <&clks IMX6QDL_CLK_MMDC_CH1_AXI>;
+	fsl,ldb-di1-parent = <&clks IMX6QDL_CLK_MMDC_CH1_AXI>;
+};
+
+&i2c2 {
+	pmic: pfuze100@08 {
+		regulators {
+			sw1a_reg: sw1ab {
+				regulator-min-microvolt = <300000>;
+				regulator-max-microvolt = <1875000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+
+			sw1c_reg: sw1c {
+				regulator-min-microvolt = <300000>;
+				regulator-max-microvolt = <1875000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+
+			sw2_reg: sw2 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw3a_reg: sw3a {
+				regulator-min-microvolt = <400000>;
+				regulator-max-microvolt = <1200000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw3b_reg: sw3b {
+				regulator-min-microvolt = <400000>;
+				regulator-max-microvolt = <1200000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw4_reg: sw4 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-boot-on;
+				regulator-always-on;
+
+			};
+
+			swbst_reg: swbst {
+				regulator-min-microvolt = <5000000>;
+				regulator-max-microvolt = <5150000>;
+				regulator-boot-on;
+				regulator-always-on;
+
+			};
+
+			snvs_reg: vsnvs {
+				regulator-min-microvolt = <1000000>;
+				regulator-max-microvolt = <3000000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vref_reg: vrefddr {
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vgen1_reg: vgen1 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1550000>;
+				regulator-always-on;
+			};
+
+			vgen2_reg: vgen2 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1550000>;
+				regulator-always-on;
+			};
+
+			vgen3_reg: vgen3 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <2800000>;
+				regulator-always-on;
+			};
+
+			vgen4_reg: vgen4 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <2500000>;
+				regulator-always-on;
+			};
+
+			vgen5_reg: vgen5 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen6_reg: vgen6 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3150000>;
+				regulator-always-on;
+			};
+		};
+	};
+
+	ov564x_mipi: ov564x_mipi@3c {  
+		DOVDD-supply = <&sw4_reg>; /* 1.8v */ 
+	};
+
+};
+&iomuxc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog>;
+
+	imx6qdl-sabresd {
+		pinctrl_hog: hoggrp {
+			fsl,pins = <
+				MX6QDL_PAD_NANDF_D0__GPIO2_IO00 0x80000000
+				MX6QDL_PAD_NANDF_D1__GPIO2_IO01 0x80000000
+				MX6QDL_PAD_NANDF_D2__GPIO2_IO02 0x80000000
+				MX6QDL_PAD_NANDF_D3__GPIO2_IO03 0x80000000
+				MX6QDL_PAD_GPIO_0__CCM_CLKO1    0x130b0
+				MX6QDL_PAD_NANDF_CLE__GPIO6_IO07 0x80000000
+				MX6QDL_PAD_NANDF_ALE__GPIO6_IO08 0x80000000
+				MX6QDL_PAD_ENET_TXD1__GPIO1_IO29 0x80000000
+				MX6QDL_PAD_EIM_D22__GPIO3_IO22  0x80000000
+				MX6QDL_PAD_ENET_CRS_DV__GPIO1_IO25 0x80000000
+				MX6QDL_PAD_EIM_D26__GPIO3_IO26 0x80000000
+				MX6QDL_PAD_EIM_CS1__GPIO2_IO24 0x80000000
+				MX6QDL_PAD_ENET_RXD0__GPIO1_IO27 0x80000000
+				MX6QDL_PAD_EIM_A25__GPIO5_IO02 0x80000000
+				MX6QDL_PAD_EIM_D23__GPIO3_IO23 0x80000000
+				MX6QDL_PAD_EIM_EB3__GPIO2_IO31 0x80000000
+				MX6QDL_PAD_SD1_CMD__GPIO1_IO18 0x80000000
+				MX6QDL_PAD_SD3_RST__GPIO7_IO08	0x80000000
+				MX6QDL_PAD_GPIO_9__GPIO1_IO09 	0x80000000
+				MX6QDL_PAD_EIM_DA9__GPIO3_IO09 0x80000000
+				MX6QDL_PAD_GPIO_1__WDOG2_B 0x80000000
+				MX6QDL_PAD_NANDF_CS0__GPIO6_IO11 0x80000000
+				MX6QDL_PAD_NANDF_CS1__GPIO6_IO14 0x80000000
+			>;
+		};
+		pinctrl_ecspi1: ecspi1grp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D17__ECSPI1_MISO 0x100b1
+				MX6QDL_PAD_EIM_D18__ECSPI1_MOSI 0x100b1
+				MX6QDL_PAD_EIM_D16__ECSPI1_SCLK 0x100b1
+			>;
+		};
+	};
+};
+
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-hdcp-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-hdcp-fix-ldo.dts
new file mode 100644
index 0000000..e0d3ce4
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-evb-hdcp-fix-ldo.dts
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2012-2014 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6dqscm-1gb-evb-fix-ldo.dts"
+
+&hdmi_video {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hdmi_hdcp>;
+	fsl,hdcp;
+};
+
+&i2c2 {
+	status = "disable";
+};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-hdcp-interleave-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-hdcp-interleave-ldo.dts
new file mode 100644
index 0000000..5625c94
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-evb-hdcp-interleave-ldo.dts
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2012-2014 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6dqscm-1gb-evb-interleave-ldo.dts"
+
+&hdmi_video {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hdmi_hdcp>;
+	fsl,hdcp;
+};
+
+&i2c2 {
+	status = "disable";
+};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-hdcp-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-hdcp-ldo.dts
deleted file mode 100644
index 2fcf4f9..0000000
--- a/arch/arm/boot/dts/imx6dqscm-1gb-evb-hdcp-ldo.dts
+++ /dev/null
@@ -1,23 +0,0 @@
-/*
- * Copyright 2012-2014 Freescale Semiconductor, Inc.
- * Copyright 2011 Linaro Ltd.
- *
- * The code contained herein is licensed under the GNU General Public
- * License. You may obtain a copy of the GNU General Public License
- * Version 2 or later at the following locations:
- *
- * http://www.opensource.org/licenses/gpl-license.html
- * http://www.gnu.org/copyleft/gpl.html
- */
-
-#include "imx6dqscm-1gb-evb-ldo.dts"
-
-&hdmi_video {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hdmi_hdcp>;
-	fsl,hdcp;
-};
-
-&i2c2 {
-	status = "disable";
-};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-ldo.dts
new file mode 100644
index 0000000..2635286
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-ldo.dts
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6q-sabresd-ldo.dts"
+
+/ {
+	model = "Freescale i.MX6D SCM EVB";
+	compatible = "fsl,imx6q-sabresd", "fsl,imx6q";
+        soc {
+            busfreq {
+                status = "disabled";
+            };
+        };
+};
+
+&ecspi1 {
+	cs-gpios = <&gpio2 30 0>;
+	internal_scm_flash: m25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "micron,n25q128a13";
+		spi-max-frequency = <20000000>;
+		reg = <0>;
+	};
+};
+
+&i2c1 {
+	ov564x: ov564x@3c {
+		DOVDD-supply = <&sw4_reg>; /* 1.8v */
+	};
+};
+
+&i2c2 {
+	pmic: pfuze100@08 {
+		regulators {
+			sw1a_reg: sw1ab {
+				regulator-min-microvolt = <300000>;
+				regulator-max-microvolt = <1875000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+
+			sw1c_reg: sw1c {
+				regulator-min-microvolt = <300000>;
+				regulator-max-microvolt = <1875000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+
+			sw2_reg: sw2 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw3a_reg: sw3a {
+				regulator-min-microvolt = <400000>;
+				regulator-max-microvolt = <1200000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw3b_reg: sw3b {
+				regulator-min-microvolt = <400000>;
+				regulator-max-microvolt = <1200000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw4_reg: sw4 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-boot-on;
+				regulator-always-on;
+
+			};
+
+			swbst_reg: swbst {
+				regulator-min-microvolt = <5000000>;
+				regulator-max-microvolt = <5150000>;
+				regulator-boot-on;
+				regulator-always-on;
+
+			};
+
+			snvs_reg: vsnvs {
+				regulator-min-microvolt = <1000000>;
+				regulator-max-microvolt = <3000000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vref_reg: vrefddr {
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vgen1_reg: vgen1 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1550000>;
+				regulator-always-on;
+			};
+
+			vgen2_reg: vgen2 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1550000>;
+				regulator-always-on;
+			};
+
+			vgen3_reg: vgen3 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <2800000>;
+				regulator-always-on;
+			};
+
+			vgen4_reg: vgen4 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <2500000>;
+				regulator-always-on;
+			};
+
+			vgen5_reg: vgen5 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen6_reg: vgen6 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3150000>;
+				regulator-always-on;
+			};
+		};
+	};
+
+	ov564x_mipi: ov564x_mipi@3c {  
+		DOVDD-supply = <&sw4_reg>; /* 1.8v */ 
+	};
+
+};
+&iomuxc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog>;
+
+	imx6qdl-sabresd {
+		pinctrl_hog: hoggrp {
+			fsl,pins = <
+				MX6QDL_PAD_NANDF_D0__GPIO2_IO00 0x80000000
+				MX6QDL_PAD_NANDF_D1__GPIO2_IO01 0x80000000
+				MX6QDL_PAD_NANDF_D2__GPIO2_IO02 0x80000000
+				MX6QDL_PAD_NANDF_D3__GPIO2_IO03 0x80000000
+				MX6QDL_PAD_GPIO_0__CCM_CLKO1    0x130b0
+				MX6QDL_PAD_NANDF_CLE__GPIO6_IO07 0x80000000
+				MX6QDL_PAD_NANDF_ALE__GPIO6_IO08 0x80000000
+				MX6QDL_PAD_ENET_TXD1__GPIO1_IO29 0x80000000
+				MX6QDL_PAD_EIM_D22__GPIO3_IO22  0x80000000
+				MX6QDL_PAD_ENET_CRS_DV__GPIO1_IO25 0x80000000
+				MX6QDL_PAD_EIM_D26__GPIO3_IO26 0x80000000
+				MX6QDL_PAD_EIM_CS1__GPIO2_IO24 0x80000000
+				MX6QDL_PAD_ENET_RXD0__GPIO1_IO27 0x80000000
+				MX6QDL_PAD_EIM_A25__GPIO5_IO02 0x80000000
+				MX6QDL_PAD_EIM_D23__GPIO3_IO23 0x80000000
+				MX6QDL_PAD_EIM_EB3__GPIO2_IO31 0x80000000
+				MX6QDL_PAD_SD1_CMD__GPIO1_IO18 0x80000000
+				MX6QDL_PAD_SD3_RST__GPIO7_IO08	0x80000000
+				MX6QDL_PAD_GPIO_9__GPIO1_IO09 	0x80000000
+				MX6QDL_PAD_EIM_DA9__GPIO3_IO09 0x80000000
+				MX6QDL_PAD_GPIO_1__WDOG2_B 0x80000000
+				MX6QDL_PAD_NANDF_CS0__GPIO6_IO11 0x80000000
+				MX6QDL_PAD_NANDF_CS1__GPIO6_IO14 0x80000000
+			>;
+		};
+		pinctrl_ecspi1: ecspi1grp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D17__ECSPI1_MISO 0x100b1
+				MX6QDL_PAD_EIM_D18__ECSPI1_MOSI 0x100b1
+				MX6QDL_PAD_EIM_D16__ECSPI1_SCLK 0x100b1
+			>;
+		};
+	};
+};
+
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-ldo.dts
deleted file mode 100644
index 2635286..0000000
--- a/arch/arm/boot/dts/imx6dqscm-1gb-evb-ldo.dts
+++ /dev/null
@@ -1,194 +0,0 @@
-/*
- * Copyright 2012 Freescale Semiconductor, Inc.
- * Copyright 2011 Linaro Ltd.
- *
- * The code contained herein is licensed under the GNU General Public
- * License. You may obtain a copy of the GNU General Public License
- * Version 2 or later at the following locations:
- *
- * http://www.opensource.org/licenses/gpl-license.html
- * http://www.gnu.org/copyleft/gpl.html
- */
-
-#include "imx6q-sabresd-ldo.dts"
-
-/ {
-	model = "Freescale i.MX6D SCM EVB";
-	compatible = "fsl,imx6q-sabresd", "fsl,imx6q";
-        soc {
-            busfreq {
-                status = "disabled";
-            };
-        };
-};
-
-&ecspi1 {
-	cs-gpios = <&gpio2 30 0>;
-	internal_scm_flash: m25p80@0 {
-		#address-cells = <1>;
-		#size-cells = <1>;
-		compatible = "micron,n25q128a13";
-		spi-max-frequency = <20000000>;
-		reg = <0>;
-	};
-};
-
-&i2c1 {
-	ov564x: ov564x@3c {
-		DOVDD-supply = <&sw4_reg>; /* 1.8v */
-	};
-};
-
-&i2c2 {
-	pmic: pfuze100@08 {
-		regulators {
-			sw1a_reg: sw1ab {
-				regulator-min-microvolt = <300000>;
-				regulator-max-microvolt = <1875000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-
-			sw1c_reg: sw1c {
-				regulator-min-microvolt = <300000>;
-				regulator-max-microvolt = <1875000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-
-			sw2_reg: sw2 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			sw3a_reg: sw3a {
-				regulator-min-microvolt = <400000>;
-				regulator-max-microvolt = <1200000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			sw3b_reg: sw3b {
-				regulator-min-microvolt = <400000>;
-				regulator-max-microvolt = <1200000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			sw4_reg: sw4 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <1800000>;
-				regulator-boot-on;
-				regulator-always-on;
-
-			};
-
-			swbst_reg: swbst {
-				regulator-min-microvolt = <5000000>;
-				regulator-max-microvolt = <5150000>;
-				regulator-boot-on;
-				regulator-always-on;
-
-			};
-
-			snvs_reg: vsnvs {
-				regulator-min-microvolt = <1000000>;
-				regulator-max-microvolt = <3000000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			vref_reg: vrefddr {
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			vgen1_reg: vgen1 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <1550000>;
-				regulator-always-on;
-			};
-
-			vgen2_reg: vgen2 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <1550000>;
-				regulator-always-on;
-			};
-
-			vgen3_reg: vgen3 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <2800000>;
-				regulator-always-on;
-			};
-
-			vgen4_reg: vgen4 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <2500000>;
-				regulator-always-on;
-			};
-
-			vgen5_reg: vgen5 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen6_reg: vgen6 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3150000>;
-				regulator-always-on;
-			};
-		};
-	};
-
-	ov564x_mipi: ov564x_mipi@3c {  
-		DOVDD-supply = <&sw4_reg>; /* 1.8v */ 
-	};
-
-};
-&iomuxc {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hog>;
-
-	imx6qdl-sabresd {
-		pinctrl_hog: hoggrp {
-			fsl,pins = <
-				MX6QDL_PAD_NANDF_D0__GPIO2_IO00 0x80000000
-				MX6QDL_PAD_NANDF_D1__GPIO2_IO01 0x80000000
-				MX6QDL_PAD_NANDF_D2__GPIO2_IO02 0x80000000
-				MX6QDL_PAD_NANDF_D3__GPIO2_IO03 0x80000000
-				MX6QDL_PAD_GPIO_0__CCM_CLKO1    0x130b0
-				MX6QDL_PAD_NANDF_CLE__GPIO6_IO07 0x80000000
-				MX6QDL_PAD_NANDF_ALE__GPIO6_IO08 0x80000000
-				MX6QDL_PAD_ENET_TXD1__GPIO1_IO29 0x80000000
-				MX6QDL_PAD_EIM_D22__GPIO3_IO22  0x80000000
-				MX6QDL_PAD_ENET_CRS_DV__GPIO1_IO25 0x80000000
-				MX6QDL_PAD_EIM_D26__GPIO3_IO26 0x80000000
-				MX6QDL_PAD_EIM_CS1__GPIO2_IO24 0x80000000
-				MX6QDL_PAD_ENET_RXD0__GPIO1_IO27 0x80000000
-				MX6QDL_PAD_EIM_A25__GPIO5_IO02 0x80000000
-				MX6QDL_PAD_EIM_D23__GPIO3_IO23 0x80000000
-				MX6QDL_PAD_EIM_EB3__GPIO2_IO31 0x80000000
-				MX6QDL_PAD_SD1_CMD__GPIO1_IO18 0x80000000
-				MX6QDL_PAD_SD3_RST__GPIO7_IO08	0x80000000
-				MX6QDL_PAD_GPIO_9__GPIO1_IO09 	0x80000000
-				MX6QDL_PAD_EIM_DA9__GPIO3_IO09 0x80000000
-				MX6QDL_PAD_GPIO_1__WDOG2_B 0x80000000
-				MX6QDL_PAD_NANDF_CS0__GPIO6_IO11 0x80000000
-				MX6QDL_PAD_NANDF_CS1__GPIO6_IO14 0x80000000
-			>;
-		};
-		pinctrl_ecspi1: ecspi1grp {
-			fsl,pins = <
-				MX6QDL_PAD_EIM_D17__ECSPI1_MISO 0x100b1
-				MX6QDL_PAD_EIM_D18__ECSPI1_MOSI 0x100b1
-				MX6QDL_PAD_EIM_D16__ECSPI1_SCLK 0x100b1
-			>;
-		};
-	};
-};
-
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-fix.dtsi b/arch/arm/boot/dts/imx6dqscm-1gb-fix.dtsi
new file mode 100644
index 0000000..4ddb0c0
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-fix.dtsi
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/ {
+	memory: memory {
+               linux,usable-memory = <0x10000000 0x20000000 0x80000000 0x20000000>; 
+        };
+};
+
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-interleave.dtsi b/arch/arm/boot/dts/imx6dqscm-1gb-interleave.dtsi
new file mode 100644
index 0000000..0497c84
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-interleave.dtsi
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/ {
+	memory: memory {
+		reg = <0x10000000 0x40000000>;
+	};
+
+};
+
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-fix-ldo.dts
new file mode 100644
index 0000000..cba0bc3
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-fix-ldo.dts
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/dts-v1/;
+
+#include "imx6q.dtsi"
+#include "imx6dqscm-qwks-rev1.dtsi"
+#include "imx6dqscm-1gb-fix.dtsi"
+/ {
+	model = "Freescale i.MX6DQ SCM QWKS";
+	compatible = "fsl,imx6q";
+};
+
+&ldb {
+	lvds-channel@0 {
+		crtc = "ipu2-di0";
+	};
+
+	lvds-channel@1 {
+		crtc = "ipu2-di1";
+	};
+};
+
+&mxcfb1 {
+	status = "okay";
+};
+
+&mxcfb2 {
+	status = "okay";
+};
+
+&mxcfb3 {
+	status = "okay";
+};
+
+&mxcfb4 {
+	status = "okay";
+};
+
+&cpu0 {
+	arm-supply = <&reg_arm>;
+	soc-supply = <&reg_soc>;
+};
+
+&gpc {
+	/* use ldo-enable, u-boot will check it and configure */
+	fsl,ldo-bypass = <0>;
+	/* watchdog select of reset source */
+	fsl,wdog-reset = <1>;
+};
+
+&wdog1 {
+	status = "okay";
+};
+
+&wdog2 {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-hdcp-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-hdcp-fix-ldo.dts
new file mode 100644
index 0000000..425db9d
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-hdcp-fix-ldo.dts
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2012-2014 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6dqscm-1gb-qwks-rev1-fix-ldo.dts"
+
+&hdmi_video {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hdmi_hdcp>;
+	fsl,hdcp;
+};
+
+&i2c2 {
+	status = "disable";
+};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-hdcp-interleave-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-hdcp-interleave-ldo.dts
new file mode 100644
index 0000000..a58ee42
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-hdcp-interleave-ldo.dts
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2012-2014 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6dqscm-1gb-qwks-rev1-interleave-ldo.dts"
+
+&hdmi_video {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hdmi_hdcp>;
+	fsl,hdcp;
+};
+
+&i2c2 {
+	status = "disable";
+};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-hdcp-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-hdcp-ldo.dts
deleted file mode 100644
index 9665d14..0000000
--- a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-hdcp-ldo.dts
+++ /dev/null
@@ -1,23 +0,0 @@
-/*
- * Copyright 2012-2014 Freescale Semiconductor, Inc.
- * Copyright 2011 Linaro Ltd.
- *
- * The code contained herein is licensed under the GNU General Public
- * License. You may obtain a copy of the GNU General Public License
- * Version 2 or later at the following locations:
- *
- * http://www.opensource.org/licenses/gpl-license.html
- * http://www.gnu.org/copyleft/gpl.html
- */
-
-#include "imx6dqscm-1gb-qwks-rev1-ldo.dts"
-
-&hdmi_video {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hdmi_hdcp>;
-	fsl,hdcp;
-};
-
-&i2c2 {
-	status = "disable";
-};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-interleave-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-interleave-ldo.dts
new file mode 100644
index 0000000..6efc0c0
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-interleave-ldo.dts
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/dts-v1/;
+
+#include "imx6q.dtsi"
+#include "imx6dqscm-qwks-rev1.dtsi"
+#include "imx6dqscm-1gb-interleave.dtsi"
+/ {
+	model = "Freescale i.MX6DQ SCM QWKS";
+	compatible = "fsl,imx6q";
+};
+
+&ldb {
+	lvds-channel@0 {
+		crtc = "ipu2-di0";
+	};
+
+	lvds-channel@1 {
+		crtc = "ipu2-di1";
+	};
+};
+
+&mxcfb1 {
+	status = "okay";
+};
+
+&mxcfb2 {
+	status = "okay";
+};
+
+&mxcfb3 {
+	status = "okay";
+};
+
+&mxcfb4 {
+	status = "okay";
+};
+
+&cpu0 {
+	arm-supply = <&reg_arm>;
+	soc-supply = <&reg_soc>;
+};
+
+&gpc {
+	/* use ldo-enable, u-boot will check it and configure */
+	fsl,ldo-bypass = <0>;
+	/* watchdog select of reset source */
+	fsl,wdog-reset = <1>;
+};
+
+&wdog1 {
+	status = "okay";
+};
+
+&wdog2 {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-ldo.dts
deleted file mode 100644
index db6baa6..0000000
--- a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-ldo.dts
+++ /dev/null
@@ -1,67 +0,0 @@
-/*
- * Copyright 2012 Freescale Semiconductor, Inc.
- * Copyright 2011 Linaro Ltd.
- *
- * The code contained herein is licensed under the GNU General Public
- * License. You may obtain a copy of the GNU General Public License
- * Version 2 or later at the following locations:
- *
- * http://www.opensource.org/licenses/gpl-license.html
- * http://www.gnu.org/copyleft/gpl.html
- */
-
-/dts-v1/;
-
-#include "imx6q.dtsi"
-#include "imx6dqscm-qwks-rev1.dtsi"
-
-/ {
-	model = "Freescale i.MX6DQ SCM QWKS";
-	compatible = "fsl,imx6q";
-};
-
-&ldb {
-	lvds-channel@0 {
-		crtc = "ipu2-di0";
-	};
-
-	lvds-channel@1 {
-		crtc = "ipu2-di1";
-	};
-};
-
-&mxcfb1 {
-	status = "okay";
-};
-
-&mxcfb2 {
-	status = "okay";
-};
-
-&mxcfb3 {
-	status = "okay";
-};
-
-&mxcfb4 {
-	status = "okay";
-};
-
-&cpu0 {
-	arm-supply = <&reg_arm>;
-	soc-supply = <&reg_soc>;
-};
-
-&gpc {
-	/* use ldo-enable, u-boot will check it and configure */
-	fsl,ldo-bypass = <0>;
-	/* watchdog select of reset source */
-	fsl,wdog-reset = <1>;
-};
-
-&wdog1 {
-	status = "okay";
-};
-
-&wdog2 {
-	status = "disabled";
-};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-wifi-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-wifi-fix-ldo.dts
new file mode 100644
index 0000000..f83145c
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-wifi-fix-ldo.dts
@@ -0,0 +1,10 @@
+/*
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6dqscm-1gb-qwks-rev1-fix-ldo.dts"
+#include "imx6dqscm-qwks-rev1-wifi.dtsi"
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-wifi-interleave-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-wifi-interleave-ldo.dts
new file mode 100644
index 0000000..311d4e0
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-wifi-interleave-ldo.dts
@@ -0,0 +1,10 @@
+/*
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6dqscm-1gb-qwks-rev1-interleave-ldo.dts"
+#include "imx6dqscm-qwks-rev1-wifi.dtsi"
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-wifi-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-wifi-ldo.dts
deleted file mode 100644
index c57a135..0000000
--- a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev1-wifi-ldo.dts
+++ /dev/null
@@ -1,10 +0,0 @@
-/*
- * Copyright (C) 2015 Freescale Semiconductor, Inc.
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- */
-
-#include "imx6dqscm-1gb-qwks-rev1-ldo.dts"
-#include "imx6dqscm-qwks-rev1-wifi.dtsi"
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-fix-ldo.dts
new file mode 100644
index 0000000..540bc8d
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-fix-ldo.dts
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/dts-v1/;
+
+#include "imx6q.dtsi"
+#include "imx6dqscm-qwks-rev2.dtsi"
+#include "imx6dqscm-1gb-fix.dtsi"
+
+/ {
+	model = "Freescale i.MX6DQ SCM QWKS";
+	compatible = "fsl,imx6q";
+};
+
+&ldb {
+	lvds-channel@0 {
+		crtc = "ipu2-di0";
+	};
+
+	lvds-channel@1 {
+		crtc = "ipu2-di1";
+	};
+};
+
+&mxcfb1 {
+	status = "okay";
+};
+
+&mxcfb2 {
+	status = "okay";
+};
+
+&mxcfb3 {
+	status = "okay";
+};
+
+&mxcfb4 {
+	status = "okay";
+};
+
+&cpu0 {
+	arm-supply = <&reg_arm>;
+	soc-supply = <&reg_soc>;
+};
+
+&gpc {
+	/* use ldo-enable, u-boot will check it and configure */
+	fsl,ldo-bypass = <0>;
+	/* watchdog select of reset source */
+	fsl,wdog-reset = <1>;
+};
+
+&wdog1 {
+	status = "okay";
+};
+
+&wdog2 {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-hdcp-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-hdcp-fix-ldo.dts
new file mode 100644
index 0000000..aaaea64c
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-hdcp-fix-ldo.dts
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2012-2014 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6dqscm-1gb-qwks-rev2-fix-ldo.dts"
+
+&hdmi_video {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hdmi_hdcp>;
+	fsl,hdcp;
+};
+
+&i2c2 {
+	status = "disable";
+};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-hdcp-interleave-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-hdcp-interleave-ldo.dts
new file mode 100644
index 0000000..bc58592
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-hdcp-interleave-ldo.dts
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2012-2014 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6dqscm-1gb-qwks-rev2-interleave-ldo.dts"
+
+&hdmi_video {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hdmi_hdcp>;
+	fsl,hdcp;
+};
+
+&i2c2 {
+	status = "disable";
+};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-hdcp-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-hdcp-ldo.dts
deleted file mode 100644
index 515f4ad..0000000
--- a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-hdcp-ldo.dts
+++ /dev/null
@@ -1,23 +0,0 @@
-/*
- * Copyright 2012-2014 Freescale Semiconductor, Inc.
- * Copyright 2011 Linaro Ltd.
- *
- * The code contained herein is licensed under the GNU General Public
- * License. You may obtain a copy of the GNU General Public License
- * Version 2 or later at the following locations:
- *
- * http://www.opensource.org/licenses/gpl-license.html
- * http://www.gnu.org/copyleft/gpl.html
- */
-
-#include "imx6dqscm-1gb-qwks-rev2-ldo.dts"
-
-&hdmi_video {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hdmi_hdcp>;
-	fsl,hdcp;
-};
-
-&i2c2 {
-	status = "disable";
-};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-interleave-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-interleave-ldo.dts
new file mode 100644
index 0000000..4b1e492
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-interleave-ldo.dts
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/dts-v1/;
+
+#include "imx6q.dtsi"
+#include "imx6dqscm-qwks-rev2.dtsi"
+#include "imx6dqscm-1gb-interleave.dtsi"
+
+/ {
+	model = "Freescale i.MX6DQ SCM QWKS";
+	compatible = "fsl,imx6q";
+};
+
+&ldb {
+	lvds-channel@0 {
+		crtc = "ipu2-di0";
+	};
+
+	lvds-channel@1 {
+		crtc = "ipu2-di1";
+	};
+};
+
+&mxcfb1 {
+	status = "okay";
+};
+
+&mxcfb2 {
+	status = "okay";
+};
+
+&mxcfb3 {
+	status = "okay";
+};
+
+&mxcfb4 {
+	status = "okay";
+};
+
+&cpu0 {
+	arm-supply = <&reg_arm>;
+	soc-supply = <&reg_soc>;
+};
+
+&gpc {
+	/* use ldo-enable, u-boot will check it and configure */
+	fsl,ldo-bypass = <0>;
+	/* watchdog select of reset source */
+	fsl,wdog-reset = <1>;
+};
+
+&wdog1 {
+	status = "okay";
+};
+
+&wdog2 {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-ldo.dts
deleted file mode 100644
index ba34058..0000000
--- a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-ldo.dts
+++ /dev/null
@@ -1,67 +0,0 @@
-/*
- * Copyright 2012 Freescale Semiconductor, Inc.
- * Copyright 2011 Linaro Ltd.
- *
- * The code contained herein is licensed under the GNU General Public
- * License. You may obtain a copy of the GNU General Public License
- * Version 2 or later at the following locations:
- *
- * http://www.opensource.org/licenses/gpl-license.html
- * http://www.gnu.org/copyleft/gpl.html
- */
-
-/dts-v1/;
-
-#include "imx6q.dtsi"
-#include "imx6dqscm-qwks-rev2.dtsi"
-
-/ {
-	model = "Freescale i.MX6DQ SCM QWKS";
-	compatible = "fsl,imx6q";
-};
-
-&ldb {
-	lvds-channel@0 {
-		crtc = "ipu2-di0";
-	};
-
-	lvds-channel@1 {
-		crtc = "ipu2-di1";
-	};
-};
-
-&mxcfb1 {
-	status = "okay";
-};
-
-&mxcfb2 {
-	status = "okay";
-};
-
-&mxcfb3 {
-	status = "okay";
-};
-
-&mxcfb4 {
-	status = "okay";
-};
-
-&cpu0 {
-	arm-supply = <&reg_arm>;
-	soc-supply = <&reg_soc>;
-};
-
-&gpc {
-	/* use ldo-enable, u-boot will check it and configure */
-	fsl,ldo-bypass = <0>;
-	/* watchdog select of reset source */
-	fsl,wdog-reset = <1>;
-};
-
-&wdog1 {
-	status = "okay";
-};
-
-&wdog2 {
-	status = "disabled";
-};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-wifi-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-wifi-fix-ldo.dts
new file mode 100644
index 0000000..8847206
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-wifi-fix-ldo.dts
@@ -0,0 +1,10 @@
+/*
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6dqscm-1gb-qwks-rev2-fix-ldo.dts"
+#include "imx6dqscm-qwks-rev1-wifi.dtsi"
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-wifi-interleave-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-wifi-interleave-ldo.dts
new file mode 100644
index 0000000..10d9d02
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-wifi-interleave-ldo.dts
@@ -0,0 +1,10 @@
+/*
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6dqscm-1gb-qwks-rev2-interleave-ldo.dts"
+#include "imx6dqscm-qwks-rev1-wifi.dtsi"
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-wifi-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-wifi-ldo.dts
deleted file mode 100644
index 4613c08..0000000
--- a/arch/arm/boot/dts/imx6dqscm-1gb-qwks-rev2-wifi-ldo.dts
+++ /dev/null
@@ -1,10 +0,0 @@
-/*
- * Copyright (C) 2015 Freescale Semiconductor, Inc.
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- */
-
-#include "imx6dqscm-1gb-qwks-rev2-ldo.dts"
-#include "imx6dqscm-qwks-rev1-wifi.dtsi"
diff --git a/arch/arm/boot/dts/imx6dqscm-2gb-evb-btwifi-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-2gb-evb-btwifi-fix-ldo.dts
new file mode 100644
index 0000000..ad1a164
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-2gb-evb-btwifi-fix-ldo.dts
@@ -0,0 +1,15 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6dqscm-1gb-evb-btwifi-fix-ldo.dts"
+#include "imx6dqscm-2gb-fix.dtsi"
+
diff --git a/arch/arm/boot/dts/imx6dqscm-2gb-evb-btwifi-interleave-ldo.dts b/arch/arm/boot/dts/imx6dqscm-2gb-evb-btwifi-interleave-ldo.dts
new file mode 100644
index 0000000..c891a3f3
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-2gb-evb-btwifi-interleave-ldo.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6dqscm-1gb-evb-btwifi-interleave-ldo.dts"
+#include "imx6dqscm-2gb-interleave.dtsi"
diff --git a/arch/arm/boot/dts/imx6dqscm-2gb-evb-btwifi-ldo.dts b/arch/arm/boot/dts/imx6dqscm-2gb-evb-btwifi-ldo.dts
deleted file mode 100644
index 9f9ccb0..0000000
--- a/arch/arm/boot/dts/imx6dqscm-2gb-evb-btwifi-ldo.dts
+++ /dev/null
@@ -1,21 +0,0 @@
-/*
- * Copyright 2012 Freescale Semiconductor, Inc.
- * Copyright 2011 Linaro Ltd.
- *
- * The code contained herein is licensed under the GNU General Public
- * License. You may obtain a copy of the GNU General Public License
- * Version 2 or later at the following locations:
- *
- * http://www.opensource.org/licenses/gpl-license.html
- * http://www.gnu.org/copyleft/gpl.html
- */
-
-#include "imx6dqscm-1gb-evb-btwifi-ldo.dts"
-
-/ {
-	memory: memory {
-		reg = <0x10000000 0x80000000>;
-	};
-
-};
-
diff --git a/arch/arm/boot/dts/imx6dqscm-2gb-evb-enetirq-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-2gb-evb-enetirq-fix-ldo.dts
new file mode 100644
index 0000000..f56c3cd
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-2gb-evb-enetirq-fix-ldo.dts
@@ -0,0 +1,15 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6dqscm-1gb-evb-enetirq-fix-ldo.dts"
+#include "imx6dqscm-2gb-fix.dtsi"
+
diff --git a/arch/arm/boot/dts/imx6dqscm-2gb-evb-enetirq-interleave-ldo.dts b/arch/arm/boot/dts/imx6dqscm-2gb-evb-enetirq-interleave-ldo.dts
new file mode 100644
index 0000000..9404128
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-2gb-evb-enetirq-interleave-ldo.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6dqscm-1gb-evb-enetirq-interleave-ldo.dts"
+#include "imx6dqscm-2gb-interleave.dtsi"
diff --git a/arch/arm/boot/dts/imx6dqscm-2gb-evb-enetirq-ldo.dts b/arch/arm/boot/dts/imx6dqscm-2gb-evb-enetirq-ldo.dts
deleted file mode 100644
index 00f9fb9..0000000
--- a/arch/arm/boot/dts/imx6dqscm-2gb-evb-enetirq-ldo.dts
+++ /dev/null
@@ -1,21 +0,0 @@
-/*
- * Copyright 2012 Freescale Semiconductor, Inc.
- * Copyright 2011 Linaro Ltd.
- *
- * The code contained herein is licensed under the GNU General Public
- * License. You may obtain a copy of the GNU General Public License
- * Version 2 or later at the following locations:
- *
- * http://www.opensource.org/licenses/gpl-license.html
- * http://www.gnu.org/copyleft/gpl.html
- */
-
-#include "imx6dqscm-1gb-evb-enetirq-ldo.dts"
-
-/ {
-	memory: memory {
-		reg = <0x10000000 0x80000000>;
-	};
-
-};
-
diff --git a/arch/arm/boot/dts/imx6dqscm-2gb-evb-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-2gb-evb-fix-ldo.dts
new file mode 100644
index 0000000..51fee26
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-2gb-evb-fix-ldo.dts
@@ -0,0 +1,15 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6dqscm-1gb-evb-fix-ldo.dts"
+#include "imx6dqscm-2gb-fix.dtsi"
+
diff --git a/arch/arm/boot/dts/imx6dqscm-2gb-evb-hdcp-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-2gb-evb-hdcp-fix-ldo.dts
new file mode 100644
index 0000000..7eaa4dc
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-2gb-evb-hdcp-fix-ldo.dts
@@ -0,0 +1,15 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6dqscm-1gb-evb-hdcp-fix-ldo.dts"
+#include "imx6dqscm-2gb-fix.dtsi"
+
diff --git a/arch/arm/boot/dts/imx6dqscm-2gb-evb-hdcp-interleave-ldo.dts b/arch/arm/boot/dts/imx6dqscm-2gb-evb-hdcp-interleave-ldo.dts
new file mode 100644
index 0000000..df68b85
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-2gb-evb-hdcp-interleave-ldo.dts
@@ -0,0 +1,15 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6dqscm-1gb-evb-hdcp-interleave-ldo.dts"
+#include "imx6dqscm-2gb-interleave.dtsi"
+
diff --git a/arch/arm/boot/dts/imx6dqscm-2gb-evb-hdcp-ldo.dts b/arch/arm/boot/dts/imx6dqscm-2gb-evb-hdcp-ldo.dts
deleted file mode 100644
index 70b37a3..0000000
--- a/arch/arm/boot/dts/imx6dqscm-2gb-evb-hdcp-ldo.dts
+++ /dev/null
@@ -1,21 +0,0 @@
-/*
- * Copyright 2012 Freescale Semiconductor, Inc.
- * Copyright 2011 Linaro Ltd.
- *
- * The code contained herein is licensed under the GNU General Public
- * License. You may obtain a copy of the GNU General Public License
- * Version 2 or later at the following locations:
- *
- * http://www.opensource.org/licenses/gpl-license.html
- * http://www.gnu.org/copyleft/gpl.html
- */
-
-#include "imx6dqscm-1gb-evb-hdcp-ldo.dts"
-
-/ {
-	memory: memory {
-		reg = <0x10000000 0x80000000>;
-	};
-
-};
-
diff --git a/arch/arm/boot/dts/imx6dqscm-2gb-evb-interleave-ldo.dts b/arch/arm/boot/dts/imx6dqscm-2gb-evb-interleave-ldo.dts
new file mode 100644
index 0000000..50e7cb5
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-2gb-evb-interleave-ldo.dts
@@ -0,0 +1,15 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "imx6dqscm-1gb-evb-interleave-ldo.dts"
+#include "imx6dqscm-2gb-interleave.dtsi"
+
diff --git a/arch/arm/boot/dts/imx6dqscm-2gb-evb-ldo.dts b/arch/arm/boot/dts/imx6dqscm-2gb-evb-ldo.dts
deleted file mode 100644
index ad8f089..0000000
--- a/arch/arm/boot/dts/imx6dqscm-2gb-evb-ldo.dts
+++ /dev/null
@@ -1,21 +0,0 @@
-/*
- * Copyright 2012 Freescale Semiconductor, Inc.
- * Copyright 2011 Linaro Ltd.
- *
- * The code contained herein is licensed under the GNU General Public
- * License. You may obtain a copy of the GNU General Public License
- * Version 2 or later at the following locations:
- *
- * http://www.opensource.org/licenses/gpl-license.html
- * http://www.gnu.org/copyleft/gpl.html
- */
-
-#include "imx6dqscm-1gb-evb-ldo.dts"
-
-/ {
-	memory: memory {
-		reg = <0x10000000 0x80000000>;
-	};
-
-};
-
diff --git a/arch/arm/boot/dts/imx6dqscm-2gb-fix.dtsi b/arch/arm/boot/dts/imx6dqscm-2gb-fix.dtsi
new file mode 100644
index 0000000..e1632d7
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-2gb-fix.dtsi
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/ {
+	memory: memory {
+               linux,usable-memory = <0x10000000 0x40000000 0x80000000 0x40000000>; 
+        };
+};
+
diff --git a/arch/arm/boot/dts/imx6dqscm-2gb-interleave.dtsi b/arch/arm/boot/dts/imx6dqscm-2gb-interleave.dtsi
new file mode 100644
index 0000000..488a55a
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dqscm-2gb-interleave.dtsi
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/ {
+	memory: memory {
+		reg = <0x10000000 0x80000000>;
+	};
+
+};
+
-- 
1.9.1

